bin/dfx-token-import

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env bash
+set -euo pipefail
+SOURCE_DIR="$(dirname "$(realpath "${BASH_SOURCE[0]}")")"
+PATH="$SOURCE_DIR:$PATH"
+. "$SOURCE_DIR/versions.bash"
+
+print_help() {
+  cat <<-EOF
+
+	Adds ledger and index canisters for a given ICRC-1 token to the local dfx.json and populates the corresponding wasm and did files.
+
+	EOF
+}
+
+# Source the clap.bash file ---------------------------------------------------
+source "$SOURCE_DIR/clap.bash"
+# Define options
+clap.define short=c long=commit desc="Commit of the IC repo to download canisters from" variable=DFX_IC_COMMIT default="$DFX_IC_COMMIT"
+clap.define short=p long=prefix desc="Prefix for the local canister names" variable=LOCAL_PREFIX default="mytoken_"
+clap.define long=check desc="Check whether the canisters have been imported" variable=DFX_CHECK nargs=0
+clap.define long=wasm_dir desc="Where to store the imported wasm files" variable=WASM_DIR default="wasms"        # Matches `dfx nns import`
+clap.define long=candid_dir desc="Where to store the imported candid files" variable=CANDID_DIR default="candid" # Matches `dfx nns import`
+# Source the output file ----------------------------------------------------------
+source "$(clap.build)"
+
+# Imports wasm, candid and populate dfx.json
+token_import() {
+  test -e dfx.json || {
+    echo "ERROR: No dfx.json found.  Please run this from inside a dfx project."
+    exit 1
+  } >&2
+
+  : Add any missing fields to dfx.json
+  # Note: The default values are placed first, so that they don't override any existing values.
+  LOCAL_PREFIX="${LOCAL_PREFIX:-}" \
+    WASM_DIR="${WASM_DIR:-}" \
+    CANDID_DIR="${CANDID_DIR:-}" \
+    jq -s '{canisters:
+      { "\(env.LOCAL_PREFIX)ledger": { build: "true", candid: "\(env.CANDID_DIR)/\(env.LOCAL_PREFIX)ledger.did", wasm: "\(env.WASM_DIR)/\(env.LOCAL_PREFIX)ledger.wasm", type: "custom" }
+      , "\(env.LOCAL_PREFIX)index" : { build: "true", candid: "\(env.CANDID_DIR)/\(env.LOCAL_PREFIX)index.did",  wasm: "\(env.WASM_DIR)/\(env.LOCAL_PREFIX)index.wasm",  type: "custom" }
+      }} * .[0]
+  ' dfx.json | sponge dfx.json
+
+  : Get the wasms
+  mkdir -p "$WASM_DIR"
+  get_wasm() {
+    local remote_name local_name local_path url
+    remote_name="$1"
```

Comment:
Stupid question, when is `remote_name` set?

Reply:
It's set on this exact line that you commented on.
It's set with the value passed as the first parameter to `get_wasm`, for example on line 55 (`ic-icrc1-ledger.wasm`).